pkg/api/customization/namespace/link.go

```diff
@@ -0,0 +1,99 @@
+package namespace
+
+import (
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/config"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/printers"
+	"strings"
+)
+
+var ExportResourcePrefixMappings = map[string]string{
+	"pods":                   "api/v1",
+	"services":               "api/v1",
+	"replicationcontrollers": "api/v1",
+	"deployments":            "apis/extensions/v1beta1",
+	"daemonsets":             "apis/extensions/v1beta1",
+	"replicasets":            "apis/extensions/v1beta1",
+	"statefulsets":           "apis/apps/v1beta1",
+	"jobs":                   "apis/batch/v1",
+	"cronjobs":               "apis/batch/v1beta1",
+}
+
+var ExportPrinters = map[string]printers.ResourcePrinter{
+	"json": &printers.JSONPrinter{},
+	"yaml": &printers.YAMLPrinter{},
+}
+
+func NewLinkHandler(next types.RequestHandler, client rest.Interface, management *config.ScaledContext) types.RequestHandler {
+	lh := &yamlLinkHandler{
+		next:       next,
+		client:     client,
```

Comment:
This client is pointed to the management server.  I assume you want a client for the users cluster.  To do the use `clustermanager.Manager` to get the client dynamically in LinkHandler.  First call Manager.ClusterName() to get the current cluster of the request, then call Manager.UserContext to get the `UserContext` object that has the UnversionedClient on it